### ISSUE TITLE
IllegalStateException in parallel consumer on error

### ISSUE THREAD

**Issue by lukestephenson** [monixio/monix#224]
I noticed during error conditions with monix Observable, I'd see many IllegalStateException logged.
```
java.lang.IllegalStateException: offer after activeCount is zero
	at monix.reactive.Consumer$LoadBalanceConsumer$AsyncQueue.offer(Consumer.scala:1113)
```

I've managed to build a small program which replicates this behaviour.  Haven't yet tracked down the cause.  The program to reproduce is:
```
import monix.eval.Task
import monix.reactive.{Consumer, Observable}

import scala.concurrent.Await
import scala.concurrent.duration.{Duration, _}

object MonixTest {
  def main(args: Array[String]): Unit = {

    implicit val scheduler = monix.execution.Scheduler.fixedPool("pool", 1)

    val iter = Iterator.range(0, 50)

    val parallelConsumer = Consumer.foreachParallelAsync[Int](5)(processAnItem)

    val future = Observable.fromIterator(iter)
      .bufferTimedWithPressure(1.second, 5)
      .concatMap(Observable.fromIterable)
      .runWith(parallelConsumer)
      .runAsync

    Await.result(future, Duration.Inf)
  }

  def processAnItem(i: Int): Task[Unit] = {
    if (i == 10) Task.raiseError(new RuntimeException("die"))
    else Task.now(())
  }
}
```

This logs the exception 4 times.

Other than noise in the logs, I don't think it causes any bad side effects.

**Comment by alexandru** [comment]
@lukestephenson that's not OK at all, that's a check meant to signal a race condition which should never happen. Unfortunately the implementation is the epitome of concurrency, effectively juggling with knives.

Will try and fix it ASAP. Thanks for the detailed report.

**Comment by alexandru** [comment]
@lukestephenson so this error is harmless after all. All the code does is to complain.

I originally feared a JVM-related concurrency problem, but I managed to reproduce the problem in tests and make sense of the error and that execution branch was legal after all and it only happens after everything gets canceled due to an error. 

It is now fix and I'm deploying `2.0.2`. Thanks for reproducing it.
